datafusion/core/src/catalog/information_schema.rs

```diff
@@ -225,34 +213,31 @@ impl InformationSchemaConfig {
 
 #[async_trait]
 impl SchemaProvider for InformationSchemaProvider {
-    fn as_any(&self) -> &(dyn Any + 'static) {
+    fn as_any(&self) -> &dyn Any {
         self
     }
 
     fn table_names(&self) -> Vec<String> {
-        vec![
-            TABLES.to_string(),
-            VIEWS.to_string(),
-            COLUMNS.to_string(),
-            DF_SETTINGS.to_string(),
-            SCHEMATA.to_string(),
-        ]
+        INFORMATION_SCHEMA_TABLES
+            .iter()
+            .map(|t| t.to_string())
+            .collect()
     }
 
     async fn table(
         &self,
         name: &str,
     ) -> Result<Option<Arc<dyn TableProvider>>, DataFusionError> {
         let config = self.config.clone();
-        let table: Arc<dyn PartitionStream> = if name.eq_ignore_ascii_case("tables") {
+        let table: Arc<dyn PartitionStream> = if name.eq_ignore_ascii_case(TABLES) {
             Arc::new(InformationSchemaTables::new(config))
-        } else if name.eq_ignore_ascii_case("columns") {
+        } else if name.eq_ignore_ascii_case(COLUMNS) {
             Arc::new(InformationSchemaColumns::new(config))
-        } else if name.eq_ignore_ascii_case("views") {
+        } else if name.eq_ignore_ascii_case(VIEWS) {
             Arc::new(InformationSchemaViews::new(config))
-        } else if name.eq_ignore_ascii_case("df_settings") {
+        } else if name.eq_ignore_ascii_case(DF_SETTINGS) {
             Arc::new(InformationSchemaDfSettings::new(config))
-        } else if name.eq_ignore_ascii_case("schemata") {
+        } else if name.eq_ignore_ascii_case(SCHEMATA) {
             Arc::new(InformationSchemata::new(config))
         } else {
             return Ok(None);
```

Comment:
nit:

```suggestion
    let table: Arc<dyn PartitionStream> = match name.to_ascii_lowercase().as_str() {
        TABLES => Arc::new(InformationSchemaTables::new(config)),
        COLUMNS => Arc::new(InformationSchemaColumns::new(config)),
        VIEWS => Arc::new(InformationSchemaViews::new(config)),
        DF_SETTINGS => Arc::new(InformationSchemaDfSettings::new(config)),
        SCHEMATA => Arc::new(InformationSchemata::new(config)),
        _ => return Ok(None),
    };
```

Reply:
Looks cleaner!

Reply:
Cannot commit this change due to "Applying suggestions on deleted lines is currently not supported.". I'll change it myself.